File: app/src/main/java/com/battlelancer/seriesguide/traktapi/MovieCheckInDialogFragment.java
package com.battlelancer.seriesguide.traktapi;

import android.os.AsyncTask;
import android.os.Bundle;
import androidx.fragment.app.FragmentManager;
import com.battlelancer.seriesguide.util.DialogTools;

/**
 * Allows to check into a movie. Launching activities should subscribe to {@link
 * TraktTask.TraktActionCompleteEvent} to display status toasts.
 */
public class MovieCheckInDialogFragment extends GenericCheckInDialogFragment {

    public static boolean show(FragmentManager fragmentManager, int movieTmdbId,
            String movieTitle) {
        MovieCheckInDialogFragment f = new MovieCheckInDialogFragment();

        Bundle args = new Bundle();
        args.putString(InitBundle.ITEM_TITLE, movieTitle);
        args.putInt(InitBundle.MOVIE_TMDB_ID, movieTmdbId);
        f.setArguments(args);

        return DialogTools.safeShow(f, fragmentManager, "movieCheckInDialog");
    }

    /**
     * Start the trakt check in task.
     */
    protected void checkInTrakt(String message) {
        new TraktTask(getContext()).checkInMovie(
                getArguments().getInt(InitBundle.MOVIE_TMDB_ID),
                getArguments().getString(InitBundle.ITEM_TITLE),
                message).executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/traktapi/MovieCheckInDialogFragment.kt
// SPDX-License-Identifier: Apache-2.0
// Copyright 2013, 2014, 2016-2018, 2022, 2023 Uwe Trottmann

package com.battlelancer.seriesguide.traktapi

import android.os.AsyncTask
import android.os.Bundle
import androidx.fragment.app.FragmentManager
import com.battlelancer.seriesguide.util.safeShow

/**
 * Allows to check into a movie. Launching activities should subscribe
 * to [TraktTask.TraktActionCompleteEvent] to display status toasts.
 */
class MovieCheckInDialogFragment : GenericCheckInDialogFragment() {

    override fun checkInTrakt(message: String) {
        @Suppress("DEPRECATION") // AsyncTask
        TraktTask(context).checkInMovie(
            requireArguments().getInt(ARG_MOVIE_TMDB_ID),
            requireArguments().getString(ARG_ITEM_TITLE),
            message
        ).executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
    }

    companion object {
        fun show(fragmentManager: FragmentManager, movieTmdbId: Int, movieTitle: String): Boolean {
            val f = MovieCheckInDialogFragment()
            val args = Bundle()
            args.putString(ARG_ITEM_TITLE, movieTitle)
            args.putInt(ARG_MOVIE_TMDB_ID, movieTmdbId)
            f.arguments = args
            return f.safeShow(fragmentManager, "movieCheckInDialog")
        }
    }
}